drivers/cc2420/cc2420.c

```diff
@@ -216,32 +217,43 @@ void cc2420_tx_exec(cc2420_t *dev)
 int cc2420_rx(cc2420_t *dev, uint8_t *buf, size_t max_len, void *info)
 {
     uint8_t len;
+    uint8_t crc_corr;
 
-    /* get the packet length (without dropping it) (first byte in RX FIFO */
+    /* get the packet length (without dropping it) (first byte in RX FIFO) */
     cc2420_ram_read(dev, CC2420_RAM_RXFIFO, &len, 1);
+
+    /* fetch and check if CRC_OK bit (MSB) is set */
+    cc2420_ram_read(dev, (CC2420_RAM_RXFIFO + len - 1), &crc_corr, 1);
```

Comment:
I'm not convinced if this reads at the right position. As a reference, have a look at the [data sheet](http://www.ti.com/lit/ds/symlink/cc2420.pdf) fig. 21. The length field is at position `CC2420_RAM_RXFIFO`. Next comes MPDU which starts at `CC2420_RAM_RXFIFO +1`, right? Then the last Byte of the MPDU (which should contain the CRC bit) would be `CC2420_RAM_RXFIFO +len`. (Note that `len` does not include the length field itself).
Anyway, even with this change I can not receive packets.